lib/composite_cache_store.rb

```diff
@@ -4,146 +4,157 @@
 require_relative "composite_cache_store/version"
 
 class CompositeCacheStore
-  DEFAULT_LAYER_1_OPTIONS = {
-    expires_in: 5.minutes,
-    size: 16.megabytes
-  }
-
-  DEFAULT_LAYER_2_OPTIONS = {
-    expires_in: 1.day,
-    size: 32.megabytes
-  }
-
-  attr_reader :layers
+  attr_reader :options, :layers
+  attr_accessor :logger
 
   # Returns a new CompositeCacheStore instance
-  def initialize(*layers)
-    if layers.blank?
-      layers << ActiveSupport::Cache::MemoryStore.new(DEFAULT_LAYER_1_OPTIONS)
-      layers << ActiveSupport::Cache::MemoryStore.new(DEFAULT_LAYER_2_OPTIONS)
-    end
+  def initialize(options = {})
+    options = options.dup || {}
+    layers = options.delete(:layers) || []
 
-    message = "All layers must be instances of ActiveSupport::Cache::Store"
-    layers.each do |layer|
-      raise ArgumentError.new(message) unless layer.is_a?(ActiveSupport::Cache::Store)
-    end
+    raise ArgumentError.new("A layered cache requires more than 1 layer!") unless layers.size > 1
 
-    layers.freeze
-    @layers = layers
+    stores = layers.select { |layer| layer.is_a? ActiveSupport::Cache::Store }
+    raise ArgumentError.new("All layers must be instances of ActiveSupport::Cache::Store!") unless stores.size == layers.size
+
+    @layers = layers.freeze
+    @logger = options[:logger]
+    @options = options
   end
 
   def cleanup(...)
-    layers.each { |store| store.cleanup(...) }
+    layers.each { |layer| layer.cleanup(...) }
   end
 
   def clear(...)
-    layers.each { |store| store.clear(...) }
+    layers.each { |layer| layer.clear(...) }
+  end
+
+  def increment(name, amount = 1, options = nil)
+    provisional_layers.each { |layer| layer.delete(name, options) }
+    layers.last.increment(name, amount, options)
   end
 
-  def decrement(...)
-    layers.each { |store| store.decrement(...) }
+  def decrement(name, amount = 1, options = nil)
+    provisional_layers.each { |layer| layer.delete(name, options) }
+    layers.last.decrement(name, amount, options)
   end
 
   def delete(...)
-    layers.each { |store| store.delete(...) }
+    layers.each { |layer| layer.delete(...) }
```

Comment:
```suggestion
    provisional_layers.each { |layer| layer.delete(...) }
    layers.last.delete(...)
```

`delete` should return true "if an entry was deleted." And I don't think the possible presence of the entry in intermediate caches, which might just have old, not-yet-consistent data, qualifies as deleting an entry.